### ISSUE TITLE
RCS1104: Consider throw expression case

### ISSUE THREAD

**Issue by anreton** [JosefPihrt/Roslynator#814]
**Product and Version Used**: 
Microsoft Visual Studio Community 2019 Version 16.10.0
Roslynator.Analyzers 3.1.0 via Nuget

**Steps to Reproduce**:
```csharp
(key is null) ? throw new ArgumentNullException(nameof(key)) : false;
```

**Actual Behavior**:
Get false diagnostic for RCS1104.

And codefix is incorrect too.
![image](https://user-images.githubusercontent.com/28877846/121266425-26cbcb00-c8c3-11eb-8ba7-4e9330d48911.png)

**Expected Behavior**:
No diagnostic for RCS1104.

https://stackoverflow.com/questions/46282957/throw-expressions-not-working-for-boolean-expressions
https://stackoverflow.com/questions/56613707/is-there-a-true-or-throw-or-non-null-or-throw-expression-in-c

**Comment by JosefPihrt** [comment]
A duplicate of #799

**Comment by JosefPihrt** [comment]
@anreton The fix is out in https://www.nuget.org/packages/Roslynator.Analyzers/3.2.0
